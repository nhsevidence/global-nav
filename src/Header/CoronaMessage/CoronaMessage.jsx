import React, { Component } from "react";
import classNames from "classnames";

import styles from "./CoronaMessage.module.scss";

class CoronaMessage extends Component {
	render() {
		const classes = classNames(styles.wrapper);

		return (
			<aside className={classes} aria-label="COVID-19 Notice">
				<div className={styles.container}>
					<p>
						Read about{" "}
						<a href="https://www.nice.org.uk/covid-19">
							our approach to COVID-19
						</a>
					</p>
				</div>
			</aside>
		);
	}
}

export default CoronaMessage;
