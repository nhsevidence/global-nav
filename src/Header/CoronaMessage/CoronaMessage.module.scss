@import '~@nice-digital/nds-core/scss/core';

.wrapper {
  background: #ffeb3b;
  padding: rem($nds-spacing-small 0);

  p {
    margin-top: 0;
  }

  a {
    @include nds-link-default;
  }

  @media print {
    display: none;
  }
}

.container {
  @include nds-container;

  > :last-child {
    margin-bottom: 0;
  }
}
