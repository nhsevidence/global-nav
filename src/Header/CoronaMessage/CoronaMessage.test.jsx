import React from "react";
import Cookies from "js-cookie";
import toJson from "enzyme-to-json";
import CoronaMessage, {
	CookieMessageVersion,
	CookieName
} from "./CoronaMessage";
import { shallow } from "enzyme";

describe("CoronaMessage", () => {
	const oldCookie = document.cookie;

	beforeEach(() => {
		// Reset overridden cookie property
		document.cookie = oldCookie;
		Cookies.set(CookieName, null);
	});

	it("Renders without crashing", () => {
		const wrapper = shallow(<CoronaMessage />);

		expect(wrapper).toHaveLength(1);
	});

	it("Doesn't render by default/on the server", () => {
		const wrapper = shallow(<CoronaMessage />, {
			disableLifecycleMethods: true
		});

		expect(wrapper.isEmptyRender()).toBe(true);
	});

	it("Renders after rehydration on the client", () => {
		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		expect(wrapper.isEmptyRender()).toBe(false);
	});

	it("Matches snapshot for new visitor", () => {
		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		expect(toJson(wrapper)).toMatchSnapshot();
	});

	it("Hides cookie message when seen correct message version", () => {
		Cookies.set(CookieName, CookieMessageVersion);

		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		expect(wrapper.isEmptyRender()).toBe(true);
	});

	it("Triggers click handler on close button click", () => {
		const handleClick = jest.spyOn(CoronaMessage.prototype, "handleClick");
		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		wrapper
			.find("button")
			.at(0)
			.simulate("click");

		expect(handleClick).toHaveBeenCalledTimes(1);
	});

	it("Hides message on close button click", () => {
		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		wrapper
			.find("button")
			.at(0)
			.simulate("click");

		expect(wrapper.isEmptyRender()).toBe(true);
	});

	it("Stores cookie on evidence domain on button click", () => {
		const oldLocation = global.window.location;
		delete global.window.location;
		global.window.location = new URL("https://www.evidence.nhs.uk/");

		const cookieSet = jest.fn();

		Object.defineProperty(document, "cookie", {
			get: jest.fn().mockImplementation(() => null),
			set: cookieSet,
			configurable: true
		});

		const wrapper = shallow(<CoronaMessage />);

		wrapper.update();

		wrapper
			.find("button")
			.at(0)
			.simulate("click");

		const date = new Date();
		date.setDate(date.getDate() + 365);

		expect(cookieSet.mock.calls[1][0]).toEqual(
			`${CookieName}=${CookieMessageVersion}; path=/; expires=${date.toUTCString()}; domain=www.evidence.nhs.uk`
		);

		// Tidy up
		global.window.location = oldLocation;
	});
});
