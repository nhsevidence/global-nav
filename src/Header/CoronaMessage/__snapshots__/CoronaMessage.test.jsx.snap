// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoronaMessage Renders correctly 1`] = `
<aside
  className="wrapper"
>
  <div
    className="container"
  >
    <p>
      Read about
       
      <a
        href="https://www.nice.org.uk/covid-19"
      >
        our approach to COVID-19
      </a>
    </p>
  </div>
</aside>
`;
